src/web/src/app/space/add/component.ts

```diff
@@ -0,0 +1,135 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { startWith, map, debounceTime, distinctUntilChanged, switchMap } from 'rxjs/operators';
+import { MatOption } from '@angular/material/core';
+import { Router, NavigationExtras } from '@angular/router';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+
+import { SpaceService } from '../service';
+import { Space } from '../model';
+import { SpaceKeyValidation } from './space-key-validator';
+import { RequestWatcherService } from '../../request-watcher/service';
+
+@Component({
+	selector: 'app-space-add',
+	templateUrl: './component.html',
+	styleUrls: ['./component.css'],
+})
+export class SpaceAddComponent implements OnInit {
+	createForm: FormGroup;
+	controlAutocomplete = new FormControl('');
+
+	spaces: Observable<Space[]>;
+	selectedSpace: Space;
+	keyWasChanged: boolean = false;
+
+	constructor(
+		private router: Router,
+		private fb: FormBuilder,
+		private space: SpaceService,
+		private spaceKeyValidator: SpaceKeyValidation,
+		private requestWatcher: RequestWatcherService
+	) {
+		this._createForm();
+	}
+
+	private translitFromRuToEn = {
+	"ё": "yo", "й": "i", "ц": "ts", "у": "u", "к": "k", "е": "e", "н": "n", "г": "g",
+	"ш": "sh", "щ": "sch", "з": "z", "х": "h", "ъ": "'", "ф": "f", "ы": "i", "в": "v", "а": "a", "п": "p", "р": "r", "о": "o", "л": "l", "д": "d", "ж": "zh", "э": "e", "я": "ya", "ч": "ch", "с": "s", "м": "m", "и": "i", "т": "t", "ь": "'", "б": "b",
+	"ю": "yu"
+	};
+
+	ngOnInit() {
+
+		this.spaces = this.controlAutocomplete.valueChanges.pipe(
+			startWith(''),
+			debounceTime(300), // Optional: debounce input changes to avoid excessive requests
+			distinctUntilChanged(), // Optional: ensure distinct values before making requests
+			switchMap(value => this.space.getList(10, 0, value || '').pipe(
+				map(res => res.results)
+			))
+		);
+
+		this.controlAutocomplete.valueChanges.pipe(
+			debounceTime(500),
+			distinctUntilChanged(),
+		).subscribe(selected => {
+			if (!selected) {
+				this.selectedSpace = null;
+			}
+		});
+	}
+
+	onSelectOption(option: MatOption) {
+		if (option) {
+			this.selectedSpace = {
+				id: option.value.id,
+				name: option.value.name,
+				key: option.value.key,
+				requestsAllowed: option.value.requestAllowed,
+				createdAt: option.value.createdAt
```

Comment:
option.value.createdAt - кажется такого значения там нет. 
```suggestion
				createdAt: null
```

Reply:
Почему нет? option.valuе в данном случае - это Space. А в Space есть createdAt.

Reply:
Точно, извини, не заметил. (
Но тогда почему не так?
```typescript
if (option?.value)
  this.selectedSpace = option.value;
```

Reply:
Сделал так, как ты предложил. Работает.